packages/dev/core/src/Maths/math.polar.ts

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import type { DeepImmutable, Nullable, float } from "../types";
+import type { Vector2, Vector3 } from "./math.vector";
+import { TmpVectors } from "./math.vector";
+
+/**
+ * Class used to store (r, theta) vector representation
+ */
+export class Polar {
+	public radius: number;
+	public theta: number;
+
+	/**
+	 * Creates a new Polar object
+	 * @param radius the radius of the vector
+	 * @param theta the angle of the vector
+	 */
+	constructor(radius: number, theta: number){
+		this.radius = radius;
+		this.theta = theta;
+	}
+
+	/**
+	 * Gets the rectangular coordinates of the current Polar
+	 * @param ref the reference to assign the result
+	 * @returns the updated reference
+	 */
+	 public toVector2ToRef(ref: Vector2): Vector2{
+		let x = this.radius * Math.cos(this.theta);
+		let y = this.radius * Math.sin(this.theta);
+		ref.set(x, y);
+		return ref;
+	}
+
+	/**
+	 * Gets the rectangular coordinates of the current Polar
+	 * @returns the rectangular coordinates
+	 */
+	public toVector2(): Vector2{
+		let ref = TmpVectors.Vector2[0];
+		return this.toVector2ToRef(ref);
+	}
+
+	/**
+	 * Converts a given Vector2 to its polar coordinates
+	 * @param v the Vector2 to convert
+	 * @param ref the reference to assign the result
+	 * @returns the updated reference
+	 */
+	public static FromVector2ToRef(v: Vector2, ref: Polar): Polar{
+		const theta  = Math.sign(v.y) * Math.acos(v.x / v.length());
+		ref.radius = v.length();
+		ref.theta = theta;
+		return ref
+	}
+
+	/**
+	 * Converts a given Vector2 to its polar coordinates
+	 * @param v the Vector2 to convert
+	 * @returns a Polar
+	 */
+	public static FromVector2(v: Vector2): Polar{
+		let polar = new Polar();
+		Polar.FromVector2ToRef(v, polar);
+		return polar;
+	}
+}
+
+
+/**
+ * Class used for (radius, theta, phi) vector representation.
+ */
+export class Spherical {
+	public radius: number;
+	public theta: number;
+	public phi: number;
+
+	constructor(radius, theta, phi){
```

Comment:
Physics convention theta is vertical angle and phi is horizontal angle.
Maths convention theta is horizontal angle and phi is vertical angle

From your formulas you have chosen the physics convention, however both conventions have the vertical angle measured from the positive Y axis to the vector not from the positive y axis so a change will be needed in the FromVector formula.